lib/src/simulations/epicycloid.dart

```diff
@@ -0,0 +1,414 @@
+                  value: innerRadius.toDouble(),
+                ),
+                Center(
+                  child: Text(
+                    "R: " + innerRadius.toString(),
+                    style: Theme.of(context).textTheme.subtitle,
+                  ),
+                ),
+                Slider(
+                  min: 0.01,
+                  max: 2,
+                  activeColor: Theme.of(context).accentColor,
+                  inactiveColor: Colors.grey,
+                  onChanged: (value) {
+                    setState(() {
+                      _scaleAmount = value;
+                    });
+                  },
+                  value: _scaleAmount,
+                ),
+                Center(
+                  child: Text(
+                    "- Scale +",
+                    style: Theme.of(context).textTheme.subtitle,
+                  ),
+                ),
+                Slider(
+                  min: 0,
+                  max: 0.1,
+                  activeColor: Theme.of(context).accentColor,
+                  inactiveColor: Colors.grey,
+                  onChanged: (value) {
+                    setState(() {
+                      f = value;
+                    });
+                  },
+                  value: f,
+                ),
+                Center(
+                  child: Text(
+                    "- Frequency +",
+                    style: Theme.of(context).textTheme.subtitle,
+                  ),
+                ),
+              ],
+            ),
+          ),
+        ),
+      ),
+      body: Container(
+        width: MediaQuery.of(context).size.width,
+        child: Stack(
+          children: <Widget>[
+            NormalEpicycloid(
+              outerRadius: outerRadius,
+              innerRadius: innerRadius,
+              f: f,
+              scaleAmount: _scaleAmount,
+              animate: animate,
+              animating: animating,
+              key: globalKey,
+              context: context,
+            ),
+            Positioned(
+              top: 5,
+              left: 5,
+              child: Visibility(
+                child: Text(
+                  'k ~ ${(innerRadius / outerRadius).toStringAsFixed(2)}',
+                  style: Theme.of(context).textTheme.subtitle,
+                ),
+              ),
+            ),
+            Positioned(
+              top: 0,
+              right: 0,
+              child: Row(
+                mainAxisAlignment: MainAxisAlignment.end,
+                children: <Widget>[
+                  Text('Animate: '),
+                  Checkbox(
+                    onChanged: (_) {
+                      setState(() {
+                        animate = !animate;
+                        if (animating) {
+                          animating = (animating && animate);
+                        }
+                      });
+                    },
+                    value: animate,
+                  ),
```

Comment:
The fill needs to be red when selected, and the button needs to be disabled when animating. Check epicycloid_curve for reference.